preproccesing.py

```diff
@@ -0,0 +1,100 @@
+import json
+import multiprocessing
+import os
+from functools import partial
+from multiprocessing import Pool
+import argparse
+import cv2
+import numpy as np
+import open3d
+import configparser
+from imageio import imwrite
+
+
+def undistort(path, fx, fy, cx, cy, k1, k2, k3, k4, k5, k6, p1, p2):
+    image_ = cv2.imread(path, -1)
+    camera_matrix = np.array([[fx, 0, cx],
+                              [0, fy, cy],
+                              [0, 0, 1]])
+    undistorted = cv2.undistort(image_, camera_matrix, np.array([k1, k2, p1, p2, k3, k4, k5, k6]))
+    return undistorted
+
+
+def xyz_to_uvz(point, fx, fy, cx, cy):
+    x, y, z = point
+    u = round(((fx * x) / z) + cx)
+    v = round(((fy * y) / z) + cy)
+    return u, v, z * 1000
+
+
+def depth_images_preprocessing(depth_images_path, fx, fy, cx, cy, k1, k2, k3, k4, k5, k6, p1, p2,
+                               width, height, width_color, height_color, transform_matrix, fx_c, fy_c, cx_c, cy_c):
+    depth_images = os.listdir(depth_images_path)
+    new_dir = os.path.join(os.path.dirname(depth_images_path), "depth_preprocessed")
+    if not os.path.isdir(new_dir):
+        os.mkdir(new_dir)
+    for image in depth_images:
+        # Depth undistortion
+        depth_undistorted = undistort(os.path.join(depth_images_path, image), fx, fy, cx, cy, k1, k2, k3, k4, k5, k6,
+                                      p1, p2)
+
+        # Point cloud creating
+        picture = open3d.geometry.Image(depth_undistorted)
+        intrinsic = open3d.camera.PinholeCameraIntrinsic(width, height, fx, fy, cx, cy)
+        pc = open3d.geometry.PointCloud()
+        extrinsic = [[1., 0., 0., 0.], [0., 1., 0., 0.], [0., 0., 1., 0.], [0., 0., 0., 1.]]
+        point_cloud = pc.create_from_depth_image(picture, intrinsic, extrinsic)
+
+        # Point cloud transformation
+        transformation_matrix = np.array(transform_matrix)
+        point_cloud = point_cloud.transform(transformation_matrix)
+
+        # Point cloud to depth image
+        img = np.zeros((height_color, width_color))
+        with Pool(multiprocessing.cpu_count()) as p:
```

Comment:
Instead of doing multiprocessing, please, formulate your calculations as matrix operations and lapack/blas under the hood of numpy will do the best optimizations

Think over, how you can represent things in form of matrix P -- stacked points and some other matrix K